### ISSUE TITLE
Run cargo with --timings option in build scripts

### ISSUE THREAD

**Issue by LogvinovLeon** [risc0/risc0#2509]
### Is there an existing issue?

- [X] I have searched the existing issues

### Experiencing problems? Have you tried our Discord first?

- [X] This is not a support question.

### Motivation

I want to be able to have quick incremental compilation.
Because risc0 build scripts don't emit cargo-rerun-if-changed - they always cause recompilation.
I do my build with --timings, but I lack insight into what's going on in inner builds:
<img width="561" alt="Screenshot 2024-11-14 at 15 36 33" src="https://github.com/user-attachments/assets/e53202fe-9300-433d-92d3-45bf49710730">

### Request

I want to understand  - what's going on within those orange 9 seconds

### Solution

Run cargo with --timings flag. It shouldn't slow down too much. Or maybe hide it behind a configuration flag

### Are you willing to help with this request?

No.

**Comment by LogvinovLeon** [comment]
Any updates on this one?

**Comment by LogvinovLeon** [comment]
Any updates on this one?

**Comment by flaub** [comment]
I'm not sure what we can do here. Do you know how the `--timings` thing works?

**Comment by LogvinovLeon** [comment]
Yes. --timings flag produces the html report of build timings. Now I can't access it as risc0 build scripts run cargo without this flag. If I had the option to pass additional cargo flags - I would be able to enable it and look into those reports analyzing my guest build performance

**Comment by flaub** [comment]
I think the time is likely taken up by our build process, which is to compute the image_id for the ELF. I'm wondering if there's a way for us to report this as part of `--timings` so I was wondering how it actually works, and less about what it does.

**Comment by LogvinovLeon** [comment]
Your build process consists of two steps
1. Running `cargo build`
2. Computing I`MAGE_ID` from `ELF`

Second one - I can't really influence other that having a smaller `ELF`.
First one - I can. I can reduce the amount of crates that are being recompiled in `guest` by rearranging the dependency tree, introducing late binding etc. In order to do those optimizations - I need the view into what's going on inside this inner build. This is what this ticket is about. I'm not asking to insert this inner knowledge into the main report. That would be ideal but probably too much work. But it would already be an improvement to have an inner report available separately. And it should be a small change. I can probably work on a PR for it if it seems useful to you.
